torch/distributed/rpc/api.py

```diff
@@ -294,6 +294,18 @@ def _init_rpc_backend(
     _set_and_start_rpc_agent(rpc_agent)
 
 
+def async_function(fn):
+    r"""
+    Wrapper of a function indicating that the return value of the function is
+    guranteed to be a ``torch.futures.Future``.
+    """
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        return fn(*args, **kwargs)
+    wrapper._wrapped_async_rpc_function = fn
```

Comment:
I am not a fan of adding one attribute to distinguish whether this is an async function. Is there a better way to achieve this? We basically just need to know if a given function is wrapped by `@rpc.async_function`.

Reply:
I'm not sure, though do we need to set this to `fn`? It looks like we just use it as a boolean check, could we just use `True` and call it `_is_async_fn`?

Reply:
It's used in the #39267 to retrieve the TorchScript function.